flink-libraries/flink-table/src/test/java/org/apache/flink/table/runtime/stream/sql/JavaSqlITCase.java

```diff
@@ -164,4 +167,35 @@ public void testUnion() throws Exception {
 
 		StreamITCase.compareWithList(expected);
 	}
+
+	@Test
+	public void testProctime() throws Exception {
+		StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+		StreamTableEnvironment tableEnv = TableEnvironment.getTableEnvironment(env);
+
+		DataStream<Tuple3<Integer, Long, String>> ds = JavaStreamTestData.getSmall3TupleDataSet(env);
+		tableEnv.registerDataStream("MyTable", ds, "a, b, c, proctime.proctime");
+
+		String sqlQuery = "select proctime from MyTable";
+
+		Table result = tableEnv.sqlQuery(sqlQuery);
+
+		tableEnv
+			.toAppendStream(result, TypeInformation.of(Row.class))
+			.addSink(new SinkFunction<Row>() {
+				@Override
+				public void invoke(Row value, Context context) throws Exception {
+
+					Timestamp procTimestamp = (Timestamp) value.getField(0);
+
+					// validate the second here
+					long procSecondTime = procTimestamp.getTime() / 1000;
```

Comment:
This causes me some trouble because cases can happen when they cross the second boundary. this is not a stable ITCase in my opinion.

Reply:
yeah, it's better to validate hour. I'll update.